### PR TITLE
[Cognitive Services - Language] Authoring -  October APIs to dev folder

### LINKED ISSUE #20053: [Cognitive Services - Language - Authoring API] <2022-10-01-preview>
Service name: Azure Cognitive Service for Language
Key contacts for the review:
- PM: Peyton Fraser <peytonfraser@microsoft.com>
- Authoring Engineering Lead: Hadeer Magdy <hadeermagdy@microsoft.com>
- [CT] Custom Summarization: Amir George <amgeo@microsoft.com>
- [CT] OOS Amir George <amgeo@microsoft.com>
- [QnA] CQA Project Versioning Victor Das <victorda@microsoft.com>
- [CLU] Versioning Amir George <amgeo@microsoft.com>
- [CT] Versioning Amir George <amgeo@microsoft.com>
- Multi region deployment Amir George <amgeo@microsoft.com>

Whether this is a new or existing API: Existing API
Whether or not your service already been deployed: Yes
A brief description of the material to be reviewed/changed: We are adding new features for the October API, including [CT] Custom Summarization, [CT] OOS, [QnA] CQA Project Versioning, [CLU] Versioning, [CT] Versioning, and Multi region deployment.

### PR DESCRIPTION
**Pull request by AmirGeorge**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).
Fix: #20053

### CODE REVIEW

File: dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json [hunk 1]
```diff
@@ -969,6 +1114,238 @@
...
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink",
+          "itemName": "value"
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/authoring/analyze-conversations/projects/{projectName}/resources/jobs/{jobId}": {
+      "get": {
```
Comment (heaths) [inline dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json:1311]: If you're using a resource LRO based on the PUT above, you should use the resource LRO pattern as described here: https://github.com/microsoft/api-guidelines/blob/vNext/azure/ConsiderationsForServiceDesign.md#put-with-additional-long-running-processing. A PUT with LRO should also return a 201, not a 202.

File: dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json [hunk 2]
```diff
@@ -1388,6 +1941,17 @@
           "items": {
             "$ref": "#/definitions/ConversationalAnalysisAuthoringExportedPrebuiltEntity"
           }
+        },
+        "regex": {
+          "description": "The regex component of the entity.",
+          "$ref": "#/definitions/ConversationalAnalysisAuthoringExportedEntityRegex"
+        },
+        "requiredComponents": {
+          "description": "The required components. Allowed values are 'learned', 'list' and 'prebuilts'.",
```
Comment (heaths) [inline dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json:1950]: I've seen this in a couple places in this PR: if values are limited, this should be an enum. Also set `modelAsString: true` to make sure any future values don't break callers.

File: dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json [hunk 3]
```diff
@@ -1732,6 +2399,33 @@
...
     },
+    "ConversationalAnalysisAuthoringExportedConversation": {
+      "description": "Represents a conversation.",
+      "required": [
+        "Location",
+        "summaryLocation"
+      ],
+      "type": "object",
+      "properties": {
+        "Location": {
```
Comment (heaths) [inline dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json:2410]: Why is `Location` capitalized? Properties should be camelCase as you've done throughout this PR.

File: dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json [hunk 4]
```diff
@@ -3010,6 +3769,14 @@
...
   },
   "parameters": {
+    "ConversationalAnalysisAuthoringArmTokenHeaderParameter": {
+      "in": "header",
+      "name": "armToken",
+      "description": "The user's ARM token.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
```
Comment (heaths) [inline dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json:3779]: The user token making the request just can't pass through? If not, other services for connected storage (BYOS) tend to pass a client ID/secret or even AAD token (which this likely is already). Or is this typical of Language? I haven't seen this pattern in Language yet.

**Review by heaths — CHANGES_REQUESTED** [review]

File: dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json (hunk 2 above)
Comment (AmirGeorge) [reply dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json:1950]: @heaths the reason string is used instead of enum is that in the future this can be extended to allow specific entity component names, which will be custom values provided by the user. Also, this is just a copy paste of what we already have in our GA 2022-05-01 APIs so we can't break it.

File: dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json [hunk 5]
```diff
@@ -969,6 +1114,238 @@
...
+            "description": "Error response.",
+            "schema": {
+              "$ref": "common.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/authoring/analyze-conversations/projects/{projectName}/resources": {
+      "put": {
```
[the remark this answers is not in the document]
Comment (AmirGeorge) [reply dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json:1155]: Last board review concluded that this should be changed to an action, which will indirectly address your comment as well.

File: dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json (hunk 1 above)
Comment (AmirGeorge) [reply dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json:1311]: Last board review concluded that this should be changed to an action, which will indirectly address your comment as well.

File: dev/cognitiveservices/data-plane/Language/analyzetext-authoring.json [hunk 6]
```diff
@@ -2431,6 +3149,13 @@
     "TextAnalysisAuthoringProjectSettings": {
       "description": "Represents the settings used to define the project behavior.",
       "type": "object",
+      "properties": {
+        "confidenceThreshold": {
```
[the remark this answers is not in the document]
Comment (AmirGeorge) [reply dev/cognitiveservices/data-plane/Language/analyzetext-authoring.json:3153]: This property already exists as "confidenceThreshold" in the May GA APis for analyze-conversations, so I guess we need to keep the same name for consistency and to avoid confusion.

File: dev/cognitiveservices/data-plane/Language/analyzetext-authoring.json [hunk 7]
```diff
@@ -869,6 +983,44 @@
         }
       }
     },
+    "/authoring/analyze-text/projects/{projectName}/models/{trainedModelLabel}/:loadSnapshot": {
```
[the remark this answers is not in the document]
Comment (AmirGeorge) [reply dev/cognitiveservices/data-plane/Language/analyzetext-authoring.json:986]: Yes the name trainedModelLabel is consistent with what we already use in the /models APIs.

File: dev/cognitiveservices/data-plane/Language/analyzetext-authoring.json [hunk 8]
```diff
@@ -354,6 +423,39 @@
         "x-ms-long-running-operation": true
       }
     },
+    "/authoring/analyze-text/projects/{projectName}/copy/jobs/{jobId}": {
+      "get": {
```
[the remark this answers is not in the document]
Comment (AmirGeorge) [reply dev/cognitiveservices/data-plane/Language/analyzetext-authoring.json:427]: Will be removed, this API won't be exposed in October.

**Comment by heaths** [comment]
@AmirGeorge any updates? /cc @peytonfraser

File: dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json (hunk 4 above)
Comment (AmirGeorge) [reply dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json]: We removed the armToken now

File: dev/cognitiveservices/data-plane/Language/analyzetext-authoring.json [hunk 9]
```diff
@@ -1801,6 +2450,52 @@
       },
       "x-ms-client-name": "ExportedClass"
     },
+    "TextAnalysisAuthoringExportedCustomAbstractiveSummarizationDocument": {
```
[the remark this answers is not in the document]
Comment (AmirGeorge) [reply dev/cognitiveservices/data-plane/Language/analyzetext-authoring.json:2450]: Synced back with the PMs and they prefered the CustomAbstractiveSummarization naming.

**Comment by AmirGeorge** [comment]
> @AmirGeorge any updates? /cc @peytonfraser

@heaths can you take a 2nd look?

**Comment by heaths** [comment]
These are still targeting 10-01-preview, which is now in specification/. Ther's also lots of checks that need to be fixed. This PR has been open for long enough you likely need to rebase on main (not merge, but rebase - ask me if you need help).

File: dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json [hunk 10]
```diff
@@ -1882,6 +1859,29 @@
       },
       "x-ms-client-name": "AssignedProjectDeploymentsMetadata"
     },
+    "ConversationalAnalysisAuthoringAssignedResourceDeploymentsMetadata": {
+      "description": "Represents the metadata for deployments assigned to a resource.",
+      "required": [
+        "value",
+        "nextLink"
```
Comment (heaths) [inline dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json:1866]: Why is this required? The last page shouldn't have it, which also means you shouldn't use `x-nullable: true`. Ideally, we should never see x-nullable.

File: dev/cognitiveservices/data-plane/Language/analyzetext-authoring.json [hunk 11]
```diff
@@ -1823,6 +1800,29 @@
       },
       "x-ms-client-name": "AssignedProjectDeploymentsMetadata"
     },
+    "TextAnalysisAuthoringAssignedResourceDeploymentsMetadata": {
+      "description": "Represents the metadata for deployments assigned to a resource.",
+      "required": [
+        "value",
+        "nextLink"
```
Comment (heaths) [inline dev/cognitiveservices/data-plane/Language/analyzetext-authoring.json:1807]: Same comment as above.

File: dev/cognitiveservices/data-plane/Language/questionanswering-authoring.json [hunk 12]
```diff
@@ -3,7 +3,7 @@
   "info": {
     "title": "Microsoft Cognitive Language Service - Question Answering - Authoring",
     "description": "The language service API is a suite of natural language processing (NLP) skills built with best-in-class Microsoft machine learning algorithms.  The API can be used to analyze unstructured text for tasks such as sentiment analysis, key phrase extraction, language detection and question answering. Further documentation can be found in <a href=\"https://docs.microsoft.com/en-us/azure/cognitive-services/text-analytics/overview\">https://docs.microsoft.com/en-us/azure/cognitive-services/text-analytics/overview</a>.",
-    "version": "2022-07-01-preview"
+    "version": "2022-10-01-preview"
```
Comment (heaths) [inline dev/cognitiveservices/data-plane/Language/questionanswering-authoring.json:6]: 2022-10-01-preview is already being published in the specification/ folder. Is this actually for 10-01-preview, or does someone need to decide on the next api-version (it's largely arbitrary) for all services?

/cc @peytonfraser @robch

File: dev/cognitiveservices/data-plane/Language/readme.md [hunk 13]
```diff
@@ -36,7 +36,7 @@ modelerfour:
   lenient-model-deduplication: true
 ```
 
-``` yaml $(tag) == 'release_2022-07-01-preview'
+``` yaml $(tag) == 'release_2022-10-01-preview'
```
Comment (heaths) [inline dev/cognitiveservices/data-plane/Language/readme.md:39]: This is already being published to specification. Either you need to target the branch used for https://github.com/Azure/azure-rest-api-specs/pull/20810, or there needs to be a new API added.

**Review by heaths — CHANGES_REQUESTED** [review]

File: dev/cognitiveservices/data-plane/Language/questionanswering-authoring.json (hunk 12 above)
Comment (AmirGeorge) [reply dev/cognitiveservices/data-plane/Language/questionanswering-authoring.json:6]: This is actually 2022-10-01-preview for Language Authoring (the change here in QnA authoring is unintended as a result of a find and replace all). Shall I target my changes to the specification folder instead?

Comment (AmirGeorge) [reply dev/cognitiveservices/data-plane/Language/questionanswering-authoring.json:6]: Nvm saw the following comment, will create a new PR into kayousef/language/2022_10_01_preview

File: dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json (hunk 10 above)
Comment (AmirGeorge) [reply dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json:1866]: The behavior we have in our previous APIs including the GA 2022-05-01 is that it is required and nullable, so for the last page it will show with a value of null. I am not sure if we can change it now or that will be considered a breaking change?

File: dev/cognitiveservices/data-plane/Language/readme.md (hunk 13 above)
Comment (AmirGeorge) [reply dev/cognitiveservices/data-plane/Language/readme.md:39]: @heaths done https://github.com/Azure/azure-rest-api-specs/pull/21029

**Comment by AmirGeorge** [comment]
Outdated PR that is not needed, the changes have already been merged in another PR to the specification folder on main:
https://github.com/Azure/azure-rest-api-specs/pull/21068
